src/cpp/src/visual_language/cdpruner/cdpruner.cpp

```diff
@@ -0,0 +1,627 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "cdpruner.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <cstring>
+#include <future>
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+#include <thread>
+
+#include "openvino/openvino.hpp"
+#include "utils.hpp"
+
+namespace ov::genai::cdpruner {
+
+namespace {
+
+bool should_log_cdpruner(bool silent, bool debug_mode) {
+    if (ov::genai::utils::env_setup_for_print_debug_info()) {
+        return true;
+    }
+    if (debug_mode) {
+        return true;
+    }
+    return !silent;
+}
+
+}  // namespace
+
```

Comment:
```suggestion
```
As agreement, logs will be in another PR. Remove this unecessary code.

Reply:
Updated.